pvextractor/geometry/path.py

```diff
@@ -255,8 +255,54 @@ def sample_polygons(self, spacing, wcs=None):
         return polygons
 
     def to_patches(self, spacing, wcs=None, **kwargs):
-        from matplotlib.patches import Polygon as MPLPolygon
-        patches = []
-        for poly in self.sample_polygons(spacing, wcs=wcs):
-            patches.append(MPLPolygon(list(zip(poly.x, poly.y)), **kwargs))
-        return patches
+        if self.width is not None:
+            from matplotlib.patches import Polygon as MPLPolygon
+            patches = []
+            for poly in self.sample_polygons(spacing, wcs=wcs):
+                patches.append(MPLPolygon(list(zip(poly.x, poly.y)), **kwargs))
+            return patches
+        else:
+            raise ValueError("Use as_artist instead")
```

Comment:
```suggestion
            raise ValueError("Use the `as_artist` function instead for paths with no defined width.")
```

Reply:
Hmm. I think this is ok, though I now see that `as_artist` is useful in both cases.